### DISCUSSION TITLE
bootloader and GAP8

### DISCUSSION THREAD

**Discussion by crazyflie-aljanabi** [bitcraze/discussions#575] [Ideas]
Hello,

 I am writing to you because I am currently having trouble downloading the bootloader onto my AI-Deck. My Task is to Control the crazyflie through the GAP8 processor and capture  the video to the computer

I have tried to follow the instructions provided in the tutorial, but I am still experiencing issues. I have connected the olimex openOCD JTAG TINY-H as instructed, but the bootloader download does not seem to be successful.

I am not sure what I am doing wrong, and I would appreciate any help or guidance that you may be able to provide. If possible, could you please provide me with some troubleshooting steps or advice that could help me resolve this issue? The informations on the webseite from bitcraze is not clear and i need your help!

Thank you for your time and assistance in this matter. I look forward to hearing back from you soon.

Best regards,
Ali Al-janabi

**Comment by krichardsson** [comment]
Hi, 
we need some more information to help you.
What exact steps are you taking?
What is happening?
What is not happening that you expect to happen?
Do you get any error messages?

**Comment by crazyflie-aljanabi** [comment]
Dear Kristoffer,


i am taking the following steps to Preparing the AI deck and Crazyflie firmware:


-  i tryied to Update Crazyflie and AIdeck firmware:



  1.  Open up the cfclient on your computer, and make sure that it is version 2022.05+ (‘Help’->’About’) [ i am not sure here which version i have to take, the version of my cfclient is: 2022.3rc2, i put this version:  version 2022.05+, right??]
  2.  Make sure that only the AI-deck is attached to the Crazyflie, with no other deck.
  3.  Go to ‘Connect’->’bootloader’
  4.  Type the address of your crazyflie, press ‘Scan’ and select your crazyflie’s URI. Make sure to choose ‘radio://…’ (not ‘usb://’). Now press ‘Connect’
  5.  In the ‘Firmware Source’ section, select 2022.05 from ‘Available downloads’. Make sure to select the right platform (cf2 is for the crazyflie 2.x ).
  6.  Press ‘Program’ and wait for the STM, NRF and ESP MCUs the have been reflashed. The crazyflie will restart a couple of times, and the flashing of the ESP (‘bcAI:esp deck memory’) takes about 3 minutes.
  7.  Once the status states ‘Idle’ and the Crazyflie is disconnected, double check if the flashing has succeded. In the cfclient, connect to the crazyflie and check in the console tab if you see: ESP32: I (910) SYS: Initialized. Also LED1 should be flashing with 2 hz. [ i didn't see: ESP32: I (910) SYS: Initialized in the console tab. I send you screenshot for that i saw in the console tab]


- Gap8 bootloader (for the AIdeck 1.0 or AIdeck 1.1 Rev A-C):

Clone, build and flash the bootloader with an Olimex ARM-USB-TINY-H JTAG using the following commands:

git clone https://github.com/bitcraze/aideck-gap8-bootloader.git
$ cd aideck-gap8-bootloader
$ docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/ftdi/olimex-arm-usb-tiny-h.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make all image flash'

but i didn't see the following:


flasher is done!
--------------------------
--------------------------
Reset CONFREG to 0
--------------------------
GAP8 examine target
RESET: jtag boot mode=3
DEPRECATED! use 'adapter [de]assert' not 'jtag_reset'

In the attached screenshot you can see the answer.

My goal is to establish a video connection to the host computer. Setting up a WiFi-based transmission of the video signal to the host computer. But i don't know how to do that, therefore i need your help please.


Best regards,

Ali Al-janabi



<https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#update-crazyflie-and-aideck-firmware>


________________________________
Von: Kristoffer Richardsson ***@***.***>
Gesendet: Mittwoch, 22. Februar 2023 09:33:49
An: bitcraze/discussions
Cc: Al-Janabi, Ali Salman Aboob; Author
Betreff: Re: [bitcraze/discussions] bootloader and GAP8 (Discussion #575)


Hi,
we need some more information to help you.
What exact steps are you taking?
What is happening?
What is not happening that you expect to happen?
Do you get any error messages?

**Reply by krichardsson** [reply]
Thanks for the clarification!

I think we can consider the first part (updating the CF and the ESP on the AI deck) to be working.

The problematic part is [flashing the GAP8 boot loader](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader-for-the-aideck-1-0-or-aideck-1-1-rev-a-c-) with the Olimex. It seems as the screenshot of the output is missing, could you add a new dump of the output please? 
Thanks!

**Comment by crazyflie-aljanabi** [comment]
Dear Kristoffer,


build and flash the bootloader with an Olimex ARM-USB-TINY-H JTAG it works now and i get the following results Screenshot attached):

flasher is done!
--------------------------
--------------------------
Reset CONFREG to 0
--------------------------
GAP8 examine target
RESET: jtag boot mode=3
DEPRECATED! use 'adapter [de]assert' not 'jtag_reset'

But i have problem with the next step "Flash Wifi Example". I get always the following issue (see the screenshot:
Reset to bootloader mode ...
Could not save cache, no writable directory
Could not save cache, no writable directory
Skipping bcAI



Without the additional issue

:esp, not in the target list
Deck bcAI:gap8, reset to bootloader
| 0% Writing to bcAI:gap8 deck memory
/ 1% Writing to bcAI:gap8 deck memory
...
\ 99% Writing to bcAI:gap8 deck memory
| 100% Writing to bcAI:gap8 deck memory


I also didn't understand the following "In your wifi list you should see ‘WiFi streaming example’. If you do, please connect to it."

How and where can i find my wifi list?

How can i run the following command:

cd examples/other/wifi-img-streamer
python opencv-viewer.py


Tell me please exactly how can i control my crazyflie throw the gap8- camera on my PC?


Thank you very much for your help!


Best regards

Ali Al-janabi

________________________________
Von: Kristoffer Richardsson ***@***.***>
Gesendet: Donnerstag, 23. Februar 2023 09:01:13
An: bitcraze/discussions
Cc: Al-Janabi, Ali Salman Aboob; Author
Betreff: Re: [bitcraze/discussions] bootloader and GAP8 (Discussion #575)


Thanks for the clarification!

I think we can consider the first part (updating the CF and the ESP on the AI deck) to be working.

The problematic part is flashing the GAP8 boot loader<https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader-for-the-aideck-1-0-or-aideck-1-1-rev-a-c-> with the Olimex. It seems as the screenshot of the output is missing, could you add a new dump of the output please?
Thanks!

**Reply by krichardsson** [reply]
Great! It sounds like the flashing now works. Correct?

You can find some old discussions [related to using wifi on the AI-deck by searching the discussions with the AI-deck label and wifi as keyword](https://github.com/orgs/bitcraze/discussions?discussions_q=label%3A%22AI+deck%22+wifi).

If this does not help, could you please open a new discussion as wifi is too far away from the title "bootloader and GAP8". Also consider marking the correct answer.
